storage/s3.go

```diff
@@ -514,6 +531,7 @@ func (s *S3) Put(
 		Key:          aws.String(to.Path),
 		Body:         reader,
 		ContentType:  aws.String(contentType),
+		Metadata:     make(map[string]*string),
```

Comment:
```suggestion
		Metadata:     make(Metadata),
```

Reply:
The [storage.Metadata](https://github.com/peak/s5cmd/blob/123b1c7fc9c614aa214a6795468fa140a38ad05e/storage/storage.go#L202) is `map[string]string`  but the [s3manager.UploadInput.Metadata](https://github.com/aws/aws-sdk-go/blob/3f477ea07655cad1462020fed4c67bfe1c555dc2/service/s3/s3manager/upload_input.go#L176) is `map[string]*string`
So they're, unfortunately of different types.